packages/components/src/Button/button.tsx

```diff
@@ -3,19 +3,19 @@ import React, { ReactNode } from "react";
 
 type ButtonHTMLElementProps = React.ButtonHTMLAttributes<HTMLButtonElement>;
 
-export type ButtonProps = {
+type ButtonPropsBase = ButtonHTMLElementProps & {
   as?: "button" | React.ComponentType<any>;
   /**
    * The button contents or label.
    */
   children: ReactNode;
   color?: ClickableProps<"button">["color"];
-  disabled?: ButtonHTMLElementProps["disabled"];
   size?: ClickableProps<"button">["size"];
-  type?: ButtonHTMLElementProps["type"];
   variant?: ClickableProps<"button">["variant"];
 };
 
+export type ButtonProps = Omit<ButtonPropsBase, "style">;
```

Comment:
We do we want to prevent people from using the style attribute? Is it to prevent inline styles? Or did we run into some issue in Learning Platform or something?

Reply:
> Is it to prevent inline styles?

Yeah, so they have to use our props and/or css modules.

> Or did we run into some issue in Learning Platform or something?

No, this isn't in response to any existing code, just thinking through what situations could a developer get into that we would want to avoid.

I'm totally open to being convinced that `style` is valid and should be allowed through.

Reply:
I don't think there's generally a use case for inline styles, although sometimes it's handy when we need an escape hatch from static CSS (even though that's what we want 99.9999999999999999% of the time).

Personally I probably wouldn't worry about preventing it's use. But it doesn't seem actively harmful to do so. If we run into a case where we want it we could always bring it back.

Reply:
Looking at the `Button` component again, I'm suddenly realizing that `className` isn't a supported prop (it is on `Clickable`). Passing in `className` (even an empty string) removes all the internal styling from the element. So uh we should probably guard against that too (unless we do actually want to support it, in which case we should add it instead).

Thoughts @dcwither ?

<img width="1270" alt="Screen Shot 2021-07-08 at 7 17 49 AM" src="https://user-images.githubusercontent.com/7761701/124937833-9f09d700-dfbc-11eb-9146-fcaa24730b44.png">

Reply:
oo good catch. I think it's better to support it than not to encourage adoption. we can change `Clickable` to destructure `className`:
```
<Component
      className={clsx(
        className,
        styles.button,
        // ... the rest
```

Reply:
actually ... now I'm wondering, what's the point of omitting `style` if we're not omitting `className` (because the latter is much much more common for styling) 🤔  should we just allow both?